src/lib/RequestParser.ts

```diff
@@ -148,28 +152,75 @@ export class RequestParser {
+                    if (!('autoTruncateMessage' in createCashlinkRequest && createCashlinkRequest.autoTruncateMessage)
+                        && didTruncate) {
+                        throw new Error('Cashlink must be shorter than 256 bytes or autoTruncateMessage must be '
+                            + 'enabled.');
+                    }
+                    message = truncated;
+                }
+
+                const theme = createCashlinkRequest.theme || CashlinkTheme.UNSPECIFIED;
+                if (!Object.values(CashlinkTheme).includes(theme) || !Nimiq.NumberUtils.isUint8(theme)) {
+                    // Also checking whether theme is a valid Uint8 to catch ids that are potentially to high in the
+                    // CashlinkTheme enum and to filter out values that are actually the enum keys that have been added
+                    // by typescript for reverse mapping.
+                    throw new Error('Invalid Cashlink theme');
                 }
+
+                const returnCashlink = !!createCashlinkRequest.returnCashlink;
+                if (returnCashlink && !isPriviledgedOrigin(state.origin)) {
+                    throw new Error(`Origin ${state.origin} is not authorized to request returnCashlink.`);
+                }
+                const skipSharing = !!createCashlinkRequest.returnCashlink && !!createCashlinkRequest.skipSharing;
+
                 return {
-                    kind: RequestType.CASHLINK,
-                    appName: cashlinkRequest.appName,
-                    cashlinkAddress: cashlinkRequest.cashlinkAddress
-                            ? Nimiq.Address.fromUserFriendlyAddress(cashlinkRequest.cashlinkAddress)
-                            : undefined,
-                    senderAddress: cashlinkRequest.senderAddress
-                            ? Nimiq.Address.fromUserFriendlyAddress(cashlinkRequest.senderAddress)
-                            : undefined,
-                    senderBalance: cashlinkRequest.senderBalance,
-                } as ParsedCashlinkRequest;
+                    kind: RequestType.CREATE_CASHLINK,
+                    appName: createCashlinkRequest.appName,
+                    senderAddress,
+                    senderBalance,
+                    value,
+                    message,
+                    theme,
+                    returnCashlink,
+                    skipSharing,
+                } as ParsedCreateCashlinkRequest;
+            case RequestType.MANAGE_CASHLINK:
+                const manageCashlinkRequest = request as ManageCashlinkRequest;
+                return {
+                    kind: RequestType.MANAGE_CASHLINK,
+                    appName: manageCashlinkRequest.appName,
+                    cashlinkAddress: manageCashlinkRequest.cashlinkAddress
+                        ? Nimiq.Address.fromUserFriendlyAddress(manageCashlinkRequest.cashlinkAddress)
+                        : undefined,
```

Comment:
In the request type, `cashlinkAddress` is not optional. So I think this should throw when no address to manage is given here. Or what is the use case?